.github/workflows/python-test-coverage.yml

```diff
@@ -30,25 +30,28 @@ jobs:
       - uses: actions/checkout@v4
       - name: Setup filename variables
         run: echo "FILE_ID=${{ github.event.number }}" >> $GITHUB_ENV
-      - name: Download coverage
-        uses: dawidd6/action-download-artifact@v6
+      - name: Download Files
+        uses: actions/download-artifact@v4
         with:
-          name: python-coverage-${{ env.FILE_ID }}.txt
-          github_token: ${{ secrets.GH_ACTIONS_PR_WRITE }}
-          workflow: python-unit-tests.yml
-          search_artifacts: true
-          if_no_artifact_found: warn
-      - name: Download pytest
-        uses: dawidd6/action-download-artifact@v6
-        with:
-          name: pytest-${{ env.FILE_ID }}.xml
-          github_token: ${{ secrets.GH_ACTIONS_PR_WRITE }}
-          workflow: python-unit-tests.yml
-          search_artifacts: true
-          if_no_artifact_found: warn
+          github-token: ${{ secrets.GH_ACTIONS_PR_WRITE }}
+          run-id: ${{ github.event.workflow_run.id }}
+          path: python/
+          merge-multiple: true
+          # workflow: python-unit-tests.yml
+          # if_no_artifact_found: ignore
+      - name: Display structure of downloaded files
+        run: ls python/
+      # - name: Download pytest
+      #   uses: dawidd6/action-download-artifact@v6
+      #   with:
+      #     name: pytest-${{ env.FILE_ID }}.xml
+      #     github_token: ${{ secrets.GH_ACTIONS_PR_WRITE }}
+      #     workflow: python-unit-tests.yml
+      #     if_no_artifact_found: ignore
       - name: Pytest coverage comment
         id: coverageComment
         uses: MishaKav/pytest-coverage-comment@main
```

Comment:
is there more official action we can use here? Something not from an individual.